src/ert/analysis/event.py

```diff
@@ -26,6 +27,20 @@ class AnalysisReportEvent(AnalysisEvent):
 
 
 @dataclass
-class AnalysisErrorEvent(AnalysisEvent):
-    smoother_snapshot: Optional[SmootherSnapshot] = None
-    error_msg: Optional[str] = None
+class AnalysisCSVEvent(AnalysisEvent):
+    name: str
+    header: List[str]
+    data: Sequence[Sequence[Union[str, float]]]
+
+    def __post_init__(self) -> None:
+        if len(self.header) != len(self.data[0]):
+            raise ValueError(
+                f"Header ({self.header}) must have same length as "
+                f"number of columns ({len(self.data[0])})"
+            )
+
+
+@dataclass
+class AnalysisErrorEvent(AnalysisCSVEvent):
+    extra: Dict[str, str]
+    error_msg: str
```

Comment:
Why is this a subclass of AnalysisCSVEvent? I think we still would like to have the one you deleted above

Reply:
It was so that I could join the `new` type of events with the existing one, and did not have to create new logic for creating a table that was very similar to the existing one, the report looks almost identical to what it did before, it just has slightly different headers. It simplified the logic in: [src/ert/gui/simulation/view/update.py](https://github.com/equinor/ert/pull/8002/files#diff-e8b14b48368b92e26182c748d0d6ade7488935a3d5678dd030b2bca1cdcd0979) a bit.

![image](https://github.com/equinor/ert/assets/44577479/dbf0e817-64b9-4c98-ad8f-6ab38d207414)